src/main/java/org/spongepowered/common/scheduler/SpongeScheduler.java

```diff
@@ -315,4 +318,18 @@ public Future<?> execute(final Runnable runnable) {
             return null;
         });
     }
+
+    protected List<ScheduledTask> drainTasks() {
+        List<ScheduledTask> list = new LinkedList<>();
+        tasks.values().removeIf(x -> {
```

Comment:
```suggestion
        this.tasks.values().removeIf(x -> {
```

Reply:
Also, how is it guaranteed this method is not going to concurrently be accessed during ticks or adding tasks? It doesn't seem threadsafe in comparison if this method gets called in a few different ways.

Reply:
This method is also thread-safe, the collections that ConcurrentHashMap returns are thread-safe, also this method has protected access, which will not allow it to be used outside, also this method is possible under race (during pool closure), but it will not break anything because before that we terminate the pool, perhaps the close method is not very thought out, but It wasn't much better